### PR TITLE
[configure-splash-screen] Add StatusBar customisation

### PR DESCRIPTION
**Pull request by bbarthec**
Follow-up of https://github.com/expo/expo/pull/8535

Partially resolves https://github.com/expo/expo/issues/7504

### CODE REVIEW

File: packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts [hunk 1]
```diff
@@ -24,17 +42,7 @@ describe('Drawable.xml', () => {
...
-  This file was created by '@expo/configure-splash-screen' and some of it's content shouldn't be modified by hand
--->
-<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
-  <item android:drawable="@color/splashscreen_background"/>
-  <item>
-    <bitmap android:gravity="center" android:src="@drawable/splashscreen_image"/>
-  </item>
-</layer-list>
-`;
+      const expected = generateDrawbleFileContent({ addBitmapItem: true });
```
Comment (tsapeta) [inline packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts:45]: Hah! I found a typo! 😁
```suggestion
      const expected = generateDrawableFileContent({ addBitmapItem: true });
```

File: packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts [hunk 2]
```diff
@@ -43,43 +51,16 @@ describe('Drawable.xml', () => {
...
       const actual = vol.readFileSync(filePath, 'utf-8');
-      const expected = `<?xml version="1.0" encoding="utf-8"?>
-<!--
-  This file was created by '@expo/configure-splash-screen' and some of it's content shouldn't be modified by hand
--->
-<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
-  <item android:drawable="@color/splashscreen_background"/>
-</layer-list>
-`;
+      const expected = generateDrawbleFileContent();
```
Comment (tsapeta) [inline packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts:54]:
```suggestion
      const expected = generateDrawableFileContent();
```

File: packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts [hunk 3]
```diff
@@ -43,43 +51,16 @@ describe('Drawable.xml', () => {
...
--->
-<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
-  <item android:drawable="@color/splashscreen_background"/>
-  <item>
-    <bitmap android:gravity="center" android:src="@drawable/splashscreen_image"/>
-  </item>
-</layer-list>
-`
-      );
+      vol.writeFileSync(filePath, generateDrawbleFileContent({ addBitmapItem: true }));
```
Comment (tsapeta) [inline packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts:60]:
```suggestion
      vol.writeFileSync(filePath, generateDrawableFileContent({ addBitmapItem: true }));
```

File: packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts [hunk 4]
```diff
@@ -43,43 +51,16 @@ describe('Drawable.xml', () => {
...
       const actual = vol.readFileSync(filePath, 'utf-8');
-      const expected = `<?xml version="1.0" encoding="utf-8"?>
-<!--
-  This file was created by '@expo/configure-splash-screen' and some of it's content shouldn't be modified by hand
--->
-<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
-  <item android:drawable="@color/splashscreen_background"/>
-</layer-list>
-`;
+      const expected = generateDrawbleFileContent();
```
Comment (tsapeta) [inline packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts:63]:
```suggestion
      const expected = generateDrawableFileContent();
```

File: packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts [hunk 5]
```diff
@@ -10,6 +10,24 @@ jest.mock('fs');
 
 describe('Drawable.xml', () => {
   describe('configureDrawableXml', () => {
+    function generateDrawbleFileContent({ addBitmapItem }: { addBitmapItem?: boolean } = {}) {
```
Comment (tsapeta) [inline packages/configure-splash-screen/src/android/__tests__/Drawable.xml.test.ts:13]:
```suggestion
    function generateDrawableFileContent({ addBitmapItem }: { addBitmapItem?: boolean } = {}) {
```

File: packages/configure-splash-screen/src/constants.ts [hunk 6]
```diff
@@ -27,3 +33,22 @@ export interface Parameters {
...
   darkModeImagePath?: string;
 }
+
+/**
+ * These might be optionally provided by the user. There are default values for them.
+ */
+export interface StatusBarOptions {
+  statusBarHidden: boolean;
+  statusBarStyle: StatusBarStyle;
+}
```
Comment (tsapeta) [inline packages/configure-splash-screen/src/constants.ts:43]: If they are optional (as said in the comment), shouldn't it be `Partial`?

File: packages/configure-splash-screen/src/constants.ts [hunk 7]
```diff
@@ -27,3 +33,22 @@ export interface Parameters {
...
+
+/**
+ * These might be optionally provided by the user. There are default values for them.
+ * Android only
+ */
+export interface AndroidOnlyStatusBarOptions {
```
Comment (tsapeta) [inline packages/configure-splash-screen/src/constants.ts:49]: Maybe `Partial` type?

File: packages/configure-splash-screen/src/constants.ts (hunk 6 above)
Comment (bbarthec) [reply packages/configure-splash-screen/src/constants.ts:43]: Not really, the second part of comment explains that there are always some value for each of these 😉

File: packages/configure-splash-screen/src/constants.ts (hunk 7 above)
Comment (bbarthec) [reply packages/configure-splash-screen/src/constants.ts:49]: Matter of discussion, but I prefer `interface` over `type` when possible 😛 Moreover, when it comes to original interface/type IMO it's better to go with explicit (via `?` modifier) than implicit via `Partial` (but matter of implicity/explicity of `?` and `Partial` is another topic for discussion)

File: packages/configure-splash-screen/CHANGELOG.md [hunk 8]
```diff
@@ -6,4 +6,6 @@
 
 ### 🎉 New features
 
+- Add StatusBar customization options. ([#2196](https://github.com/expo/expo-cli/pull/2196) by [@bbarthec](https://github.com/bbarthec))
```
[the remark this answers is not in the document]
Comment (bbarthec) [reply packages/configure-splash-screen/CHANGELOG.md:9]: Ok, I believe it's better to keep it consistent across the `expo/expo-cli` repo 😉 Sooo, no `CHANGELOG` here, at least for now 🥳

**Pull request merged**